### DISCUSSION TITLE
Azure Remote-backed Storage configuration fails in bootstrap: repository type [azure] does not exist

### DISCUSSION THREAD

**Discussion by paulmossman** [Opster/opensearch-k8s-operator#886] [General]
Hi.  I'm trying to configure Azure Remote-backed Storage via the operator, which is installed via Helm Chart.

I configured it, but upon deploy the bootstrap Pod fails with:
```
[ERROR][o.o.b.OpenSearchUncaughtExceptionHandler] [cluster-bootstrap-0] uncaught exception in thread [main]
org.opensearch.bootstrap.StartupException: RepositoryException[[remote-backed-storage] repository type [azure] does not exist]
```

So I configured spec.bootstrap.pluginsList to ["repository-azure"] as per [User Guide - Adding plugins](https://github.com/opensearch-project/opensearch-k8s-operator/blob/main/docs/userguide/main.md#adding-plugins), but the same error happens.

I see that the pluginsList and keystore fields were added to BootstrapConfig on Sept 19th under [commit fb3812d](https://github.com/opensearch-project/opensearch-k8s-operator/commit/fb3812d58795838607bbfaeefd6670d3762a71f3), and this isn't available in 2.6.1.  (`k explain OpenSearchCluster.spec.bootstrap.pluginsList` results in `error: field "pluginsList" does not exist`.)

Is it possible to configure Azure Remote-backed Storage via the released version of the operator?  If not, then when is the next release planned for?

**Comment by swoehrl-mw** [comment]
Hi @paulmossman.
There is a workaround described in https://github.com/opensearch-project/opensearch-k8s-operator/issues/430#issuecomment-1432818721.

I believe @prudhvigodithi is planning a release soon, so hopefully this feature will be released within the next few weeks.

**Reply by paulmossman** [reply]
Thank you @swoehrl-mw!  The workaround does allow the bootstrap node to complete.  But then the first cluster node fails with `a remote store node ... is trying to join a non remote store cluster`.

But I see that @prudhvigodithi released v2.7.0, and using that I'm able to set up Remote-backed Storage.  (Note that `bootstrap:` needs `pluginsList: ["repository-azure"]` and `keystore: ...`, but not `additionalConfig`.)

**Reply by swoehrl-mw** [reply]
Thanks for the feedback @paulmossman. Interesting error. But since it works with the newly released version now I'll not look into that further.

**Comment by paulmossman** [comment]
Understood @swoehrl-mw, thank you.
